src/core/observables/LBProfileObservable.hpp

```diff
@@ -57,19 +57,19 @@ class LBProfileObservable : public ProfileObservable {
     sampling_positions.clear();
     if (sampling_delta_x == 0 or sampling_delta_y == 0 or sampling_delta_z == 0)
       throw std::runtime_error("Parameter delta_x/y/z must not be zero!");
-    const auto n_samples_x = static_cast<size_t>(std::rint(
-        (box_geo.length()[0] - sampling_offset_x) / sampling_delta_x));
-    const auto n_samples_y = static_cast<size_t>(std::rint(
-        (box_geo.length()[1] - sampling_offset_y) / sampling_delta_y));
-    const auto n_samples_z = static_cast<size_t>(std::rint(
-        (box_geo.length()[2] - sampling_offset_z) / sampling_delta_z));
+    const auto n_samples_x =
+        static_cast<size_t>(std::rint((max_x - min_x) / sampling_delta_x));
+    const auto n_samples_y =
+        static_cast<size_t>(std::rint((max_y - min_y) / sampling_delta_y));
+    const auto n_samples_z =
+        static_cast<size_t>(std::rint((max_z - min_z) / sampling_delta_z));
```

Comment:
I don't understand the logic here. Let's take the observable `LBVelocityProfile` as an example. We sample the LB velocity every `sampling_delta_*`, but the histogram stores the result in an array of dimensions `n_*_bins`. This is why you had zeroes on the last dimension of each coordinate: there is no guarantee that the allocated histogram has the same dimensions as the ones calculated here. You could even get into a segfault if you don't correctly guess the values and give too small `n_*_bins` sizes. Am I missing something here?

Reply:
the histogram is independent of the LB fluid sampling. There might be the case of no hits for a bin. That's why there is a flag ```allow_empty_bins```.

Reply:
alright, thanks!